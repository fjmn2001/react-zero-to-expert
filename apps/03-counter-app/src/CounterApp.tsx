import "./index.css"
import { useState } from "react"

const CounterApp = ({ value = 0 }: { value?: number }) => {
  const [counter, setCounter] = useState(value)

  const handleIncrement = () => {
    setCounter((counter) => counter + 1)
  }
  const handleReset = () => {
    setCounter(() => value)
  }
  const handleDecrement = () => {
    setCounter((counter) => counter - 1)
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handleIncrement}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleDecrement}>-1</button>
    </>
  )
}

export default CounterApp
