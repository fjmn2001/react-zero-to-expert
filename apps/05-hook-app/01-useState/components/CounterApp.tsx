import "./counter.css"
import { useState } from "react"

const CounterApp = () => {
  const [state, setState] = useState({
    counter1: 10,
    counter2: 20,
  })
  const { counter1, counter2 } = state
  const handleIncrement = () =>
    setState((counter) => ({
      ...counter,
      counter1: counter1 + 1,
    }))

  return (
    <>
      <h1>Counter {counter1}</h1>
      <h1>Counter {counter2}</h1>
      <hr />
      <button onClick={handleIncrement} className={"btn btn-primary"}>
        +1
      </button>
    </>
  )
}

export default CounterApp
