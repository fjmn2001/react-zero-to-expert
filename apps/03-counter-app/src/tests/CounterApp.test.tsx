import { describe, expect, test } from "vitest"
import { shallow } from "enzyme"
import CounterApp from "../CounterApp"

describe("Testing <CounterApp/>", () => {
  test("it should show CounterApp", () => {
    const wrapper = shallow(<CounterApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test("it should show CounterApp and 100 by default", () => {
    const value = 100
    const wrapper = shallow(<CounterApp value={value} />)
    const defaultValue = wrapper.find("h2").text()

    expect(`${value}`).toBe(defaultValue)
  })
})
