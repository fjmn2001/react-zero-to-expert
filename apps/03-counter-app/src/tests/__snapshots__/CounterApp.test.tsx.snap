// Vitest Snapshot v1

exports[`Testing <CounterApp/> > it should show CounterApp 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
